### PR TITLE
sorter/leveldb: add cleaner

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Add cleaner to clean moved or stopped tables' data.

### Check List 

Tests 

 - Unit test

```
ok      github.com/pingcap/ticdc/cdc/sorter/leveldb     4.437s  coverage: 87.1% of statements
ok      github.com/pingcap/ticdc/cdc/sorter/leveldb/message     0.106s  coverage: 100.0% of statements
```

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/actor/actor.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ type Mailbox interface {
 
 	// Try to receive a message.
 	// It must be nonblocking and should only be called by System.
-	tryReceive() (message.Message, bool)
+	Receive() (message.Message, bool)
```
Comment (sdojjy) [inline pkg/actor/actor.go:59]: Base on the comment, TryReceive is better?

Comment (overvenus) [reply pkg/actor/actor.go:59]: I'll update comment, as we have `Send` for nonblocking send, we should name `Receive` for nonblocking receive.

**Review by asddongmen — APPROVED** [review]

File: cdc/sorter/leveldb/cleaner.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+	limiter *rate.Limiter
+	router  *actor.Router
+}
+
+var _ actor.Actor = (*CleanerActor)(nil)
+
+// NewCleanerActor returns a cleaner actor.
+func NewCleanerActor(
+	ctx context.Context, id int, db *leveldb.DB, router *actor.Router,
```
Comment (hi-rustin) [inline cdc/sorter/leveldb/cleaner.go:49]: It looks like we have some parameters that are not used. Are they reserved parameters?

File: cdc/sorter/leveldb/cleaner.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+		start := encoding.EncodeTsKey(task.UID, task.TableID, 0)
+		limit := encoding.EncodeTsKey(task.UID, task.TableID+1, 0)
+		iterRange := &lutil.Range{
+			Start: start,
+			Limit: limit,
+		}
+		iter := clean.db.NewIterator(iterRange, nil)
+
+		// Force write the first batch if the task is rescheduled (rate limited).
```
Comment (hi-rustin) [inline cdc/sorter/leveldb/cleaner.go:108]:
```suggestion
		// Force writes the first batch if the task is rescheduled (rate limited).
```

File: cdc/sorter/leveldb/cleaner.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+					rescheduleDelay = delay
+					break TASKS
+				}
+				batch.Reset()
+				force = false
+			}
+		}
+		// Release iterator in time.
+		iter.Release()
+		// Ignore rate limter and force write remaining kv.
```
Comment (hi-rustin) [inline cdc/sorter/leveldb/cleaner.go:134]:
```suggestion
		// Ignore rate limiter and force write remaining kv.
```

File: cdc/sorter/leveldb/cleaner.go (hunk 2 above)
Comment (overvenus) [reply cdc/sorter/leveldb/cleaner.go:49]: No, nice catch!

Comment (hi-rustin) [reply cdc/sorter/leveldb/cleaner.go:49]: Do we need to keep the ctx?

**Comment by hi-rustin** [comment]
@overvenus check failed.

File: cdc/sorter/leveldb/cleaner.go (hunk 2 above)
Comment (overvenus) [reply cdc/sorter/leveldb/cleaner.go]: Addressed, thanks!

File: cdc/sorter/leveldb/cleaner.go [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+	closedWg *sync.WaitGroup
+
+	limiter *rate.Limiter
+	router  *actor.Router
+}
+
+var _ actor.Actor = (*CleanerActor)(nil)
+
+// NewCleanerActor returns a cleaner actor.
+func NewCleanerActor(
```
Comment (amyangfei) [inline cdc/sorter/leveldb/cleaner.go:48]: Is leveldb sorter and cleaner shared by all changefeeds in one TiCDC process or each changefeed has a unique one?

**Review by hi-rustin — APPROVED** [review]

File: cdc/sorter/leveldb/cleaner.go (hunk 5 above)
Comment (overvenus) [reply cdc/sorter/leveldb/cleaner.go:48]: They are shared by all changefeeds in one TiCDC process for easy resource management. And they can be shared for each changefeed if we want, no additional modification required.

Comment (amyangfei) [reply cdc/sorter/leveldb/cleaner.go:48]: If shared by multiple changefeeds, what if one changefeed trigger cleaner and cleans data that another changefeed will use?

Comment (overvenus) [reply cdc/sorter/leveldb/cleaner.go:48]: Data is isolated by a UID prefix.

**Pull request merged**